forcedotcom/dev-scripts
# BREAKING CHANGE: bump husky to v7

### What does this PR do?
Bumps husky to v7.

husky breaking changes: https://typicode.github.io/husky/#/?id=breaking-changes

**QA**:

1. Clone a repo that's using dev-scripts like [plugin-source](https://github.com/salesforcecli/plugin-source)
2. Instead of `yarn link`ing, reference to dev-scripts locally [in this line](https://github.com/salesforcecli/plugin-source/blob/458964beb3088bd6248767d500196b86b370689b/package.json#L23): `"@salesforce/dev-scripts": "../dev-scripts"`
3. Run `yarn install`, you should see husky being upgraded and and error asking to run `yarn install` again to finish ⬇️ 
4. Run `yarn install` from the previous step, you should see the `.husky` field in plugin-source's package.json removed and a new `.husky` folder created with 3 hooks.
5. Try to commit a change and verify that all git hooks works.

**Roll out**
I'm leaning towards a manual roll out, mainly because:

1. [husky doesn't autoinstall git hooks anymore](https://blog.typicode.com/husky-git-hooks-autoinstall/), the recommendation is to handle this in a `prepare` script, but every package using `dev-scritps` already uses `prepare` to run `sf-install`.  I've added a some code in `sf-install` to run `husky install` but there are some repos that:

    1.  Uses `dev-scripts` to handle deps/scripts AND 🔽 
    2. Doesn't have `prepare` set to `sf-install` (easy to fix), or already use `prepare` for specific things (have to manually edit).

2. After husky 7 is installed, you have to add and commit the `.husky` folder created in the package repo.



### What issues does this PR fix or reference?
@W-10088025@
